scapy/packet.py

```diff
@@ -993,19 +1102,21 @@ def loop(todo, done, self=self):
         return loop(todo, done)
 
     def __iterlen__(self):
+        # type: () -> int
         """Predict the total length of the iterator"""
         fields = [key for (key, val) in itertools.chain(six.iteritems(self.default_fields),  # noqa: E501
                   six.iteritems(self.overloaded_fields))
                   if isinstance(val, VolatileValue)] + list(self.fields)
         length = 1
 
         def is_valid_gen_tuple(x):
+            # type: (Any) -> bool
```

Comment:
Or literally any other value? The code calling it is, in summary:
```
for field in fields:
    fld, val = self.getfield_and_val(field)
    if is_valid_gen_tuple(val):
        [....]
```
 so it could really be anything :/